app/(main)/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -153,22 +166,105 @@ describe('Week', () => {
     });
   });
 
-  test('should not display GlobalSpinner after loading data', async () => {
+  test('should display main content after data is loaded and hide GlobalSpinner', async () => {
     mockUseAuthContext.isSignedIn = true;
-    mockCreateWeeklyPicks.mockResolvedValue({});
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
 
     render(
       <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
     );
-    await waitFor(() => {
-      expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
-    });
+
+    // Wait for the main content to be displayed
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
+  });
+
+  test('should not show previous weeks picks if there is no history of selected teams', async () => {
+    mockUseAuthContext.isSignedIn = true;
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
+
+    render(
+      <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
+    );
+
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    const userPickHistory = screen.queryByTestId('user-pick-history');
+    expect(userPickHistory).not.toBeInTheDocument();
+  });
+
+  test('should show previous weeks picks if their are picks in selected teams history', async () => {
```

Comment:
After confirming with Shashi that we're using `it`, please change this

Reply:
✅ done!